app/controllers/api/providers_controller.rb

```diff
@@ -50,8 +51,27 @@ def delete_resource(type, id = nil, _data = nil)
       end
     end
 
+    def custom_attributes_edit_resource(object, type, id, data = nil)
+      format_provider_custom_attributes(data)
```

Comment:
It might be better if this wasn't destructive: 

```ruby
formatted_data = format_provider_custom_attributes(data)
super(object, type, id, formatted_data)
```